src/sardana/taurus/core/tango/sardana/macro.py

```diff
@@ -1191,44 +1262,109 @@ def createMacroNode(macro_name, params_def, macro_params):
     param_nodes = macro_node.params()
-    # validate if creating XML is possible (two last cases) 
-    contain_param_repeat = False
-    len_param_nodes = len(param_nodes)
-    for i, param_node in enumerate(param_nodes):
-        if isinstance(param_node, RepeatParamNode):
-            if contain_param_repeat:
-                msg = "Only one repeat parameter is allowed"
-                raise Exception(msg)
-            if i < len_param_nodes - 1:
-                msg = "Repeat parameter must be the last one"
-                raise Exception(msg)
-            contain_param_repeat = True
-    # this ignores raw_parameters which exceeds the param_def  
-    for param_node, param_raw in zip(param_nodes, macro_params):
-        if isinstance(param_node, SingleParamNode):
-            param_node.setValue(param_raw)
-        # the rest of the values are interpreted as repeat parameter
-        elif isinstance(param_node, RepeatParamNode):
-            params_info = param_node.paramsInfo()
-            params_info_len = len(params_info)
-            rep = 0; mem = 0
-            rest_raw = macro_params[i:]
-            for member_raw in rest_raw:
-                repeat_node = param_node.child(rep)
-                # add a new repeat node (this is needed when the raw values
-                # fill more repeat nodes that the minimum number of 
-                # repetitions e.g. min=0
-                if repeat_node is None:
-                    repeat_node = param_node.addRepeat()
-                member_node = repeat_node.child(mem)
-                if isinstance(member_node, RepeatParamNode):
-                    msg = ("Nested repeat parameters are not allowed")
+
+    param_nodes_copy = copy.copy(param_nodes)
+    # Check if ParamRepeat used in advanced interface
+    open_bracks = 0
+    close_bracks = 0
+    new_interface = False
+    for param in macro_params:
+        if param.find('[') > -1:
+            open_bracks = open_bracks + 1
+        if param.find(']') > -1:
+            close_bracks = close_bracks + 1
+
+    # open_bracks and close_bracks don't have to be equal because several closes or opens can be in the same param
+    if open_bracks and close_bracks:
+        new_interface = True
+
+    # Close the character strings into ''
+    if new_interface:       
+        # Create a string
+        params_str = ''
+        for par in macro_params:
+            params_str = params_str + str(par) + ","
+        params_str = params_str[:-1] 
+        params_str = params_str.replace("[,", "[").replace(",]","]").replace("][", "],[")
+        macro_params = eval(params_str, globals(), Reflector())
+        if type(macro_params) == list:
+            new_params = []
+            new_params.append(macro_params)
+            macro_params = new_params
+        # Convert tuple to list
+        macro_params = list(macro_params)
+        macro_params = recur_map(str, macro_params)
+
+    
+    if not new_interface:
```

Comment:
Hi Zibi, in the lines above we are not working with the macro_node but only with the parameters, so it has not sense for me to implement something in the macro_node class only for covering that conversion (since it has still nothing to do with the macro_node). What do you
meant with that?. Thanks.